### PR TITLE
Video pdf refactor

### PR DESCRIPTION
**Pull request by Jwan622**
-added video model because we eventually need to store youtube urls
-added video tests
-fixed pdf model to take out the page column
-fixed pdf tests

### CODE REVIEW

File: app/controllers/admin/pdfs_controller.rb [hunk 1]
```diff
@@ -14,6 +14,6 @@ def create
   end
 
   def pdf_params
-    params.require(:pdf).permit(:pdf_file, :url, :page, :slot)
+    params.require(:pdf).permit(:pdf_file, :url, :slot)
```
Comment (cluhring) [inline app/controllers/admin/pdfs_controller.rb:17]: these look strong

File: app/helpers/youtube_helper.rb [hunk 2]
```diff
@@ -6,3 +12,14 @@ def embed(youtube_url)
...
 end
+
+# #equiv
+# YoutubeHelper.extend(YoutubeHelper)
+#
+# #equiv
+# extend self
+#
+# #equiv
+# YoutubeHelper.singleton_class.class_eval do
```
Comment (cluhring) [inline app/helpers/youtube_helper.rb:23]: how relevent

File: app/models/video.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class Video < ActiveRecord::Base
+  validate :verify_youtube_video
+
+  validates :slot, numericality: { only_integer: true,
+                                   greater_than_or_equal_to: 1,
+                                   less_than_or_equal_to: 4
+                                  },
+                                  allow_blank: true
+
+  validates_inclusion_of :page, in: ["tools and resources",
```
Comment (cluhring) [inline app/models/video.rb:10]: video needs to know its page?  why?

File: app/models/video.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+class Video < ActiveRecord::Base
+  validate :verify_youtube_video
+
+  validates :slot, numericality: { only_integer: true,
```
Comment (cluhring) [inline app/models/video.rb:4]: only 4 slots available to put a video?  is that what this is saying?

File: spec/models/video_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    video4 = build(:video, page:"learn more")
+    video5 = build(:video, page:"home")
+
+    expect(video1).to be_valid
+    expect(video2).to be_invalid
+    expect(video3).to be_invalid
+    expect(video4).to be_valid
+    expect(video5).to be_valid
+  end
+end
```
Comment (cluhring) [inline spec/models/video_spec.rb:50]: in last test, 2 and 3 are invalid, because only one per page?

File: app/models/video.rb (hunk 3 above)
Comment (Jwan622) [inline app/models/video.rb:10]: There are many videos on different pages and so I think when you create a video, you should be able to designate what page it's on?

File: spec/models/video_spec.rb (hunk 5 above)
Comment (Jwan622) [inline spec/models/video_spec.rb:50]: They are invalid because they have the wrong page name.

File: app/models/video.rb (hunk 4 above)
Comment (Jwan622) [inline app/models/video.rb:4]: Yes.

File: app/controllers/admin/pdfs_controller.rb (hunk 1 above)
Comment (oorion) [inline app/controllers/admin/pdfs_controller.rb:17]: Are we mixing pdf and video attributes here?  I thought a video has a url but a pdf does not.  If a pdf should have a url, I don't see the url in the schema.

File: app/helpers/youtube_helper.rb [hunk 6]
```diff
@@ -1,4 +1,10 @@
 module YoutubeHelper
+  # Look below for expanded code
+  class << YoutubeHelper
+    include YoutubeHelper
```
Comment (oorion) [inline app/helpers/youtube_helper.rb:4]: this looks slightly different from what was merged in your embed pr.

File: app/models/video.rb (hunk 3 above)
Comment (oorion) [inline app/models/video.rb:10]: Wouldn't this be specified by its slot?  I'm not sure though.

File: app/models/video.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+                                      "home",
+                                      "techhire locations",
+                                      "start hiring"],
+                                message: "%{value} is not a valid page",
+                                allow_blank: true
+
+  def verify_youtube_video
+    yt_regex = /www\.youtube\.com\/watch\?v=([a-zA-Z0-9_-]*)/
+
+    if self.url.blank?
```
Comment (oorion) [inline app/models/video.rb:21]: I don't think we need to specify self here.

File: app/models/video.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+      errors[:error] << "Please enter a valid url."
+    elsif !self.url.match(yt_regex)
```
Comment (oorion) [inline app/models/video.rb:23]: or here.

File: app/views/static_pages/home.html.erb [hunk 9]
```diff
@@ -9,5 +9,5 @@
 <div class="map"></div>
 
 <div class="embed-container">
-  <%= embed("https://www.youtube.com/watch?v=ZzQ3eBerHfM") %>
+  <%= YoutubeHelper.embed("https://www.youtube.com/watch?v=ZzQ3eBerHfM") %>
```
Comment (oorion) [inline app/views/static_pages/home.html.erb:12]: I like that you made it more explicit!  :sparkler:

File: spec/models/video_spec.rb [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+  it "should be invalid with a blank url" do
+    video = build(:video, url: "")
+    video.save
+
+    expect(video.errors.full_messages.to_sentence).to eql("Error Please enter a valid url.")
+    expect(video).to be_invalid
+  end
+
+  it "should ve valid with a lot of 1,2,3 or 4" do
```
Comment (oorion) [inline spec/models/video_spec.rb:18]: *slot

File: spec/models/video_spec.rb [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  it "should be invalid with an invalid page name" do
+    video = build(:video, page:"zzzzz")
+
+    expect(video).to be_invalid
+    expect(video.errors).not_to be_empty
+  end
+
+  it "should be valid with a valid page name" do
```
Comment (oorion) [inline spec/models/video_spec.rb:37]: Nice tests!!!

File: app/controllers/admin/pdfs_controller.rb (hunk 1 above)
Comment (Jwan622) [inline app/controllers/admin/pdfs_controller.rb:17]: You bring up a good point. I will take this out of my current PR.

File: app/models/video.rb (hunk 3 above)
Comment (Jwan622) [inline app/models/video.rb:10]: A slot in my mind is a position on a specific page, not a position on the site.

File: app/models/video.rb (hunk 7 above)
Comment (Jwan622) [inline app/models/video.rb:21]: You are correct but I am always unsure about where self is needed.

File: app/views/static_pages/home.html.erb (hunk 9 above)
Comment (Jwan622) [inline app/views/static_pages/home.html.erb:12]: hurrah!

**Pull request merged**